### ISSUE TITLE
Do not expose /run/utmp to users.

### ISSUE THREAD

**Issue by shakna-israel** [hashbang/hashbang#25]
So, I ran ```ls``` on one of my directories, and I got an odd response.

```
s4b3r6@sf1 ~/Projects/Siteful % ls
templ.sh
user has logged off pts/194 from 50-83-46-94
```

I replaced the user name there, and I got a few more 'logged off' messages after the first.

**Comment by shakna-israel** [comment]
Ah!

I think I see what's happening here:

WeeChat is active on my current terminal, but I also have shell access.

(As I'm getting logged on/logged off events, but still have zsh access).

**Comment by KellerFuchs** [comment]
Yes, that's a standard ZSH feature, controlled by the grml parameter [`watch`](https://grml.org/zsh/grmlzshrc.html):
>As in tcsh(1) an array of login/logout events to be reported by the shell
builtin "log". For details see zshparam(1). Defaults to (notme root).

The data is, IIRC, obtained from `/run/utmp`, like `who(1)` does.
If this is an issue, we can change the permissions on `/run/utmp`, but this is no “data leak”, in the sense that pretty much all UNIX systems expose that information.

**Comment by KellerFuchs** [comment]
Actually, let's change this.

**Comment by daurnimator** [comment]
We used to `chmod -r` it, but something kept changing it back. I think maybe some pam plugin?

See
  - https://github.com/hashbang/shell-etc/issues/84#issuecomment-205133202
  - https://github.com/hashbang/shell-etc/commit/c1f190ae0005fd972fa1c25d38a116e42ab640d2

**Comment by KellerFuchs** [comment]
`/run` is a `tmpfs`, so `/run/utmp` is recreated at every boot.

**Comment by daurnimator** [comment]
> /run is a tmpfs, so /run/utmp is recreated at every boot.

I thought it was /var/run/utmp. When did that change?

**Comment by daurnimator** [comment]
> I thought it was /var/run/utmp. When did that change?

Ah ha! /var/run is a symlink to /run. I didn't know that.
So yes, it's just reverting because it's on a tmpfs.
We should just be able to create a systemd-tmpfiles snippet for it?

**Comment by KellerFuchs** [comment]
Yes, that would be the idea but I didn't have time (yet) to test on a local machine.
